### PR TITLE
Fix design of `InlineActionsContainer` - resolv design review comments

### PR DESCRIPTION
**Pull request by susnux**
### 📝 Summary

* Resolves design review comments of @nimishavijay https://github.com/nextcloud/text/pull/3530#issuecomment-1343955035

Items within the container are now sized equally and use the whole width.
Icons are centered within the actions as suggested in the comment.

I decided to do it using a more generic approach, being reactive on the default slot, so the component can be reused more easily.

#### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
buttons use 44px fixed width | buttons with 33% width

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [x] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [x] Documentation ([README](https://github.com/nextcloud/text/blob/master/README.md) or [documentation](https://github.com/nextcloud/documentation/blob/master/admin_manual/configuration_server/text_configuration.rst#L2)) has been updated or is not required

### CODE REVIEW

File: src/components/InlineActionsContainer.vue [hunk 1]
```diff
@@ -19,7 +19,7 @@
   -->
 <template>
 	<li class="inline-container-base">
-		<ul>
+		<ul class="inline-container-content">
 			<slot />
 		</ul>
 	</li>
```
Comment (skjnldsv) [inline src/components/InlineActionsContainer.vue:25]: Could we have that component exposed and documented? :)

File: src/components/InlineActionsContainer.vue [hunk 2]
```diff
@@ -34,17 +34,45 @@ import { defineComponent } from 'vue'
...
+		 *
+		 * @param {Node} target The mutation target
+		 */
+		onMutation(target) {
+			const children = [...target.childNodes].filter(node => node instanceof HTMLElement)
+			children.forEach(node => {
+				node.style.width = `${Math.round(100 / children.length)}%`
+			})
+		},
+	},
```
Comment (skjnldsv) [inline src/components/InlineActionsContainer.vue:62]: This seems overly complicated?
Let me suggest a css only fix

File: src/components/InlineActionsContainer.vue [hunk 3]
```diff
@@ -34,17 +34,45 @@ import { defineComponent } from 'vue'
...
-	}
+ul.inline-container-content {
+	display: flex;
+	justify-content: space-between;
+
 	.action-button {
+		// Fix action buttons beeing shifted to the left (right padding)
 		padding: 0 !important;
+		// We force a with other than 44px so we need to center the items again
+		justify-content: center;
```
Comment (skjnldsv) [inline src/components/InlineActionsContainer.vue:75]:
```suggestion
ul.inline-container-content {
	display: flex;
	justify-content: space-between;
	li {
		flex: 1 1;
	}

	.action-button {
		// Fix action buttons beeing shifted to the left (right padding)
		padding: 0 !important;
		width: 100%;
		display: flex;
		justify-content: center;
```

**Review by skjnldsv — CHANGES_REQUESTED** [review]

File: src/components/InlineActionsContainer.vue (hunk 2 above)
Comment (skjnldsv) [reply src/components/InlineActionsContainer.vue:62]: https://github.com/nextcloud/text/pull/3544#discussion_r1044635494

**Review by nimishavijay — COMMENTED** [review]
Looks great design-wise! :) 🚀

**Review by nimishavijay — APPROVED** [review]

File: src/components/InlineActionsContainer.vue (hunk 2 above)
Comment (susnux) [reply src/components/InlineActionsContainer.vue]: Thank you!

File: src/components/InlineActionsContainer.vue (hunk 1 above)
Comment (susnux) [reply src/components/InlineActionsContainer.vue:25]: Submitting to nextcloud vue components?

**Review by max-nextcloud — APPROVED** [review]

File: src/components/InlineActionsContainer.vue (hunk 1 above)
Comment (skjnldsv) [reply src/components/InlineActionsContainer.vue:25]: Yes or something like that

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**